azext_iot/tests/iothub/configurations/test_iot_config_unit.py

```diff
@@ -899,3 +899,40 @@ def test_config_apply_edge_error(
                 hub_name=mock_target["entity"],
                 content=sample_config_edge_malformed,
             )
+
+
+class TestConfigTestQueries:
+    @pytest.fixture(params=[200])
+    def serviceclient(self, mocker, fixture_ghcs, fixture_sas, request):
+        service_client = mocker.patch(path_service_client)
+        service_client.return_value = build_mock_response(mocker, request.param, {})
+        return service_client
+
+    def test_config_queries(self, fixture_cmd, serviceclient):
+        subject.iot_hub_configuration_test_queries(
+            cmd=fixture_cmd,
+            hub_name=mock_target["entity"],
+            target_condition="tags.building=43 and tags.environment='test'",
+            custom_metric_queries='{"hi": "sd"}',
+        )
+        args = serviceclient.call_args
+        url = args[0][0].url
+        method = args[0][0].method
+        body = args[0][2]
+
+        assert "/configurations/testQueries" in url
+        assert method == "POST"
+
+        assert body.get("targetCondition") == "tags.building=43 and tags.environment='test'"
+        assert body.get("customMetricQueries") == {'hi': 'sd'}
+
+    def test_config_queries_invalid_args(
+        self, fixture_cmd, serviceclient
+    ):
+        with pytest.raises(CLIError):
```

Comment:
we should start checking what error message it throws too but that can be a future pr.